src/main/scala/is/hail/distributedmatrix/KeyedIndexedRowMatrix.scala

```diff
@@ -0,0 +1,29 @@
+package is.hail.distributedmatrix
+
+import is.hail.utils._
+import org.apache.spark.mllib.linalg.distributed.IndexedRowMatrix
+import org.json4s.jackson
+
+case class KeyedIndexedRowMatrix(irm: IndexedRowMatrix, keys: Option[Array[Any]], partStarts: Option[Array[Long]]) {
```

Comment:
What's this about?  Is it temporary?  This seems like it's going in the wrong direction but maybe I don't understand.

Reply:
I was experimenting with how to pass the partStarts through to write. I'm also not super happy with it.